# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1291789207 18000
#      Wed Dec 08 01:20:07 2010 -0500
# Node ID baeeb6bc6afee5b4ed64e7f156de40fc40cc71f5
# Parent  f8e97e9a9301bb537c41085d3d05509252b7d2d9
stop trying to parse file if out_of_date_check encounters an error

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-12-08  John W. Eaton  <jwe@octave.org>
+
+	* symtab.cc (symbol_table::fcn_info::fcn_info_rep::find_user_function):
+	Return if an error occurs in out_of_date_check.
+
 2010-12-07  John W. Eaton  <jwe@octave.org>
 
 	* symtab.cc (symbol_table::fcn_info::fcn_info_rep::find): If an
 	error occurred during the first call to xfind, don't try again.
 
 2010-12-07  John W. Eaton  <jwe@octave.org>
 
 	* oct-parse.yy (case_list): Also accept default_case by itself.
diff --git a/src/symtab.cc b/src/symtab.cc
--- a/src/symtab.cc
+++ b/src/symtab.cc
@@ -967,17 +967,17 @@ symbol_table::fcn_info::fcn_info_rep::fi
 octave_value
 symbol_table::fcn_info::fcn_info_rep::find_user_function (void)
 {
   // Function on the path.
 
   if (function_on_path.is_defined ())
     out_of_date_check (function_on_path);
 
-  if (! function_on_path.is_defined ())
+  if (! error_state || function_on_path.is_defined ())
     {
       std::string dir_name;
 
       std::string file_name = load_path::find_fcn (name, dir_name);
 
       if (! file_name.empty ())
         {
           octave_function *fcn = load_fcn_from_file (file_name, dir_name);
